app/components/ui/create-tale-modal/component.js

```diff
@@ -0,0 +1,158 @@
+      this.set('dataSet', []);
+      this.set('title', null);
+      $('.ui.dropdown.compute-environment').dropdown('clear');
+    },
+
+    // ---------------------------------------------------------------------------------
+    // If being referred to from a third party, then prepopulate the modal from query params
+    // ---------------------------------------------------------------------------------
+    setModalFromQueryParams() {
+      try {
+        if (this.queryParams.uri) {
+          this.set('importing', true);
+          let {api, name, uri, environment} = this.queryParams;
+
+          this.set('dataSet', uri);
+          this.set('title', name || uri);
+
+          let {official, nonOfficial} = this.computeEnvironments;
+          let image = official.findBy('name', environment) || nonOfficial.findBy('name', environment);
+          if (!image) throw new Error('Could not find specified environment: '+environment);
+
+          this.set('imageId', image.id);
+
+          $('.ui.dropdown.compute-environment').dropdown('set selected', image.id);
+
+          this.set('datasetAPI', api);
+        }
+      } catch(e) {
+        later(()=>$('.ui.modal.create-tale').modal('hide'), 50);
+        this.handleError({responseJSON:{message:e+""}});
+      }
+    },
+
+    // ---------------------------------------------------------------------------------
+    // Sets the default action to take when clicking on "Create New Tale" button
+    // Choices are: Create Tale   ,    Create Tale and Launch
+    // ---------------------------------------------------------------------------------
+    setDefaultAction(createAndLaunch) {
+      this.set('createAndLaunch', createAndLaunch);
+    }
+  },
+
+  // ---------------------------------------------------------------------------------
+  // Display error message
+  // ---------------------------------------------------------------------------------
+  handleError(e) {
+    let errorMessage = (e.responseJSON ? e.responseJSON.message : this.get('defaultErrorMessage'));
+    this.set('errorMessage', errorMessage);
+    later(()=>$('.ui.compose-error.modal').modal('show'), 100);
+  },
+
+  // ---------------------------------------------------------------------------------
+  // Calls POST /tale
+  // if launch is false: transitions to the "tale view" pane on success
+  // if launch is true:  launches the tale and transitions to the "run" pane on success
+  // ---------------------------------------------------------------------------------
+  async createNewTale(title, imageId, dataSet, config) {
+    const store = this.get('store');
+    let newTale = store.createRecord('tale', {title, imageId, dataSet, config});
+
+    try {
+      let item = await newTale.save();
+      let taleId = item.id;
+      
+      if (this.createAndLaunch) {
+        let newInstance = this.get('store').createRecord('instance');
+        newInstance.save({adapterOptions:{queryParams:{imageId, taleId}}});
```

Comment:
```suggestion
        await newInstance.save({adapterOptions:{queryParams:{imageId, taleId}}});
```

Reply:
No. It does not. await is used if you want to capture the result of the promise. We are not capturing anything here.